### DISCUSSION TITLE
v2.0.0 - 完全进化！

### DISCUSSION THREAD

**Discussion by Steve-xmh** [amll-dev/applemusic-like-lyrics#114] [Announcements]
经过两个月的近乎不间断爆肝，2.0.0 大幅度重构版本终于做的差不多了！

## 更新内容

完全重写的播放页面！以近乎匹配 iPad 版本的 Apple Music！

![image](https://user-images.githubusercontent.com/39523898/230704176-610d4a64-b33f-4d3e-934a-4387bd9c81f5.png)

英文单词的辉光效果！

![image](https://user-images.githubusercontent.com/39523898/230705314-44f28376-c3d8-4119-8482-f441565574db.png)

右键菜单！提供各种功能，复制歌词，查看歌曲信息还有更多！

![image](https://user-images.githubusercontent.com/39523898/230705888-1036ee05-a0f1-4652-aee9-b50eccab5edb.png)

播放页面内的插件设置！同时尽可能自制了 AppKit 风格的组件库！

![image](https://user-images.githubusercontent.com/39523898/230704231-950b12ce-7523-46cf-bad6-e1973218c213.png)

> 注：你仍然可以在 BetterNCM 的插件管理器中调节插件设置

音频可视化效果！你可以将播放控制栏更换成音频可视化组件，跟着音乐节奏一起摇摆吧！

> 注：该功能需要你安装 LibFrontendPlay 并启用或使用 MRBNCM 插件加载器方可使用，否则默认将隐藏该功能。

![image](https://user-images.githubusercontent.com/39523898/230704540-1b45fdab-be21-4285-a0f6-5abca75103e5.png)

实验性支持 TTML 格式歌词！此格式为 Apple Music 内部使用的 Syllable Lyric 逐拍歌词文件格式，可以提供更加详细的歌词信息和逐词信息，能够帮助 AMLL 给你展示更加出色的歌词效果！
（如副唱歌词靠右（对唱效果），背景人声歌词等效果）

> 注：现在没有比较好的途径获取 TTML 歌词，你可以通过在 https://github.com/Steve-xmh/amll-ttml-db 的 Issues 中请愿你需要 TTML 格式的歌曲，由大家为你抓取相关文件哦！
> 同时我还有一个用于编辑抓取 TTML 歌词的工具仍在制作中，也欢迎大家前来旁观哦：https://github.com/Steve-xmh/amll-ttml-tool

![image](https://user-images.githubusercontent.com/39523898/230704426-c145733e-52a5-472a-8084-91c418e4cc43.png)
![image](https://user-images.githubusercontent.com/39523898/230704437-9dd0a7f3-adcb-42e4-a19e-797e8abceb74.png)

重做间奏三点呼吸动画！行为将会和 iPad 版本的 Apple Music 一致，根据间曲长度调节呼吸速度！

![image](https://user-images.githubusercontent.com/39523898/230705064-11474ea6-3a7a-4d67-ae80-26850b143f4e.png)

## 其它细节/问题修复

- 修正了页面内调节背景时不能更新采样色的问题 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/109
- 完成了歌曲信息悬浮滚动效果 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/75 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/76
- 修正播放模式卡死问题 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/104
- 三点大小不再跟随自定义歌词字体大小
- 翻新了关于页面，调节了右键菜单的项目顺序
- 增加了循环切换播放模式按钮 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/101
- 增加了心型喜欢/取消喜欢歌曲按钮 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/102
- 歌词贡献信息分割线可以隐藏了 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/100
- 现在可以配置左右侧播放控制按钮的行为了（解决 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/81 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/80 需求）
- 修复音质标签对齐问题 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/95 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/99
- 修复播放暂停按钮在初次打开网易云时失效的问题 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/94
- 增加了复制歌词源文件功能 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/88
- 增加了创作者和歌词贡献者 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/73
- 优化了文字可读性 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/72
- 修正了动画时长为负值的问题 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/92
- 初步增加三点呼吸动画效果
- 增加了交错歌词行的缓冲
- 对多个 CJK 字符组成的单词进行拆分播放以保证渐变效果一致
- 合并标点符号单词以免影响歌词行时长计算
- 给播放页面内设置窗口的红绿灯加了图标
- 增加了提前歌词行时序功能
- 修正了播放按钮问题
- 优化了流体背景效果
- 修正了取色失败率高的问题
- 增加了不缓存歌词数据状态
- 让歌词页面的其他字体颜色跟随 AMLL 字体设置
- 修正歌词行组件销毁时没有断开大小变更监听器导致卡顿的问题
- 修正了自定歌词写入了错误的文件夹的问题
- 修正了无歌词的样式和提示信息
- 优化了歌词位置的定位
- 修正专辑名字体颜色样式
- 其它细节优化

## 最后

感谢在这期间帮我贡献各类 Bug 和功能建议的用户们！

<hr /><em>This discussion was created from the release <a href='https://github.com/Steve-xmh/applemusic-like-lyrics/releases/tag/v2.0.0'>v2.0.0 - 完全进化！</a>.</em>

**Comment by gfzzip** [comment]
好耶！！！

**Comment by Dos7t3r** [comment]
太强了大佬！

**Comment by 225hhh666-tech** [comment]
厉害
